check_out_of_date.py

```diff
@@ -0,0 +1,186 @@
+    package_map = dict()
+    for pkg, ver in pkgs:
+        if pkg not in package_map:
+            package_map[pkg] = [ver]
+        else:
+            package_map[pkg].append(ver)
+    return package_map
+
+
+def exec_command_strict(cmd):
+    """ Wraps check_output splitting the input and string'ing the output"""
+    return bytes.decode(subprocess.check_output(cmd.split()))
+
+
+def exec_command(cmd):
+    """ Like exec_command_strict but returns the output even if the command exited unsuccessfully"""
+    try:
+        return exec_command_strict(cmd)
+    except subprocess.CalledProcessError as e:
+        return bytes.decode(e.output)
+
+
+def get_portage_tree_packages(tree_path):
+    """ returns a list of all packages in a portage tree/overlay in the form of cat/pkg-ver"""
+    pkgs = exec_command_strict("find -L {} -maxdepth 3 -type f -name *.ebuild -not -name skel.ebuild -printf %P\\n".format(tree_path))
+
+    def process_line(line):
+        # cat/pkg/pkg-ver.ebuild -> cat/pkg-ver
+        chunks = line.split("/")
+        end = chunks[2].replace(".ebuild", "")
+        return chunks[0] + "/" + end
+    return build_pkg_map(map(process_line, pkgs.splitlines()))
+
+
+def process_emerge_output(eout):
+    """ transform from emerge --unordered-dispaly to cat/pkg-ver"""
+    def process_line(line):
+        return line.strip().split("] ")[1].split(":")[0]
+
+    def is_package(line):
+        # none of the header line have a /
+        return "/" in line
+
+    return map(process_line, filter(is_package, eout.splitlines()))
+
+
+def get_board_packages(board):
+    """ gets a list of packages used by a board. valid boards are {arm,amd}64-usr, sdk, and bootstrap"""
+    emerge_args = "--emptytree --pretend --verbose --unordered-display"
+    if board == "sdk":
+        cmd = "emerge {} @system sdk-depends sdk-extras".format(emerge_args)
+    elif board == "amd64-usr" or board == "arm64-usr":
+        cmd = "emerge-{} {} @system board-packages".format(board, emerge_args)
+    elif board == "bootstrap":
+        pkgs = exec_command_strict("/usr/lib64/catalyst/targets/stage1/build.py")
+        cmd = "emerge {} {}".format(emerge_args, pkgs)
+    else:
+        raise "invalid board"
+    return build_pkg_map(process_emerge_output(exec_command(cmd)))
+
+
+def print_table(report, head, line_head, line_tail, tail, joiner, pkg_joiner):
+    print(head)
+    # metapackage that acts as the header
+    report.insert(0, {"name": "Package",
+                      "common": ["Common"],
+                      "ours": ["Ours"],
+                      "upstream": ["Upstream"],
+                      "tag": "Tag",
+                      "sdk": ["sdk"],
+                      "arm64-usr": ["arm64-usr"],
+                      "amd64-usr": ["amd64-usr"],
+                      "bootstrap": ["bootstrap"],
+                      "modified": "Modified"})
+    for entry in report:
+        print(line_head + joiner.join([entry["name"],
```

Comment:
Best practices would still have you use a `get` function even though this will always be defined in the context of this script. If this script was imported and used by something else it would error out with a `KeyError` if any of these fields were not defined.